### PR TITLE
Fix area around bottom tab avatar isn't clickable

### PR DESCRIPTION
**Pull request by bernhardoj**
### Details

The are around the bottom tab avatar isn't clickable.

### Fixed Issues

$ https://github.com/Expensify/App/issues/38369
PROPOSAL: https://github.com/Expensify/App/issues/38369#issuecomment-1998877292

### Tests

Same as QA Steps
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as QA Steps
### QA Steps

A.
1. Click/press around the area of the bottom tab avatar
2. Verify that it opens the settings page

B.
1. Click/press around the area of the bottom tab FAB
2. Verify the FAB menu is visible
3. Click/press again around the area of the bottom tab FAB
4. Verify the FAB menu closes
5. Hover over the area of all the 3 bottom tab menus (Chats, FAB, and Profile)
6. Verify the tooltip shows
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/50919443/cb2b9528-8e3f-466e-8bfc-ce1a6bc8d09a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/50919443/756ec51a-c97c-4409-b3b4-77180be3c692

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/50919443/20e6ca72-d489-466c-bf76-3c5744e340da

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/50919443/6680ce8c-e92e-4cc2-9035-157baf8fddb1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/50919443/efb51d73-0961-428e-8ae9-5c9dff925cbc

https://github.com/Expensify/App/assets/50919443/ef89caad-dff0-4dd9-abee-acc83cb14836

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/50919443/4b2f189f-510d-4344-a1cb-5e53e799ca36

https://github.com/Expensify/App/assets/50919443/33899e3a-58ff-44b4-b13a-38e6180135fe

</details>

### CODE REVIEW

**Review by paultsimura — CHANGES_REQUESTED** [review]

**Comment by mountiny** [comment]
@bernhardoj yep if you can fix that here too happy with $500

**Comment by bernhardoj** [comment]
Found the issue. When we press the area around FAB, `isCreateMenuActive` value is false.

https://github.com/Expensify/App/blob/e9c112a2b89217401dfef3e248fa8e93d4c4e418/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js#L142-L147

That is because, in PopoverProvider, we have a click listener that will close a popover, except if the popover `anchorRef` is the same as the element that we click.

https://github.com/Expensify/App/blob/e9c112a2b89217401dfef3e248fa8e93d4c4e418/src/components/PopoverProvider/index.tsx#L37-L44

FAB currently has 2 pressable and the `anchorRef` is set to the inner pressable.
https://github.com/Expensify/App/blob/e9c112a2b89217401dfef3e248fa8e93d4c4e418/src/components/FloatingActionButton.tsx#L106-L120

So, if we press the outer pressable, the PopoverProvider will close the FAB menu then the FAB click (`toggleCreateMenu`) will be executed which will show the popover again.

We can fix it by only having the outer pressable and replacing the inner one with an animated view, but the problem that comes after that is the tooltip won't hide when the popover is visible.

https://github.com/Expensify/App/assets/50919443/a3e4b58a-4feb-4d16-9a8c-3b7b62ce5c7f

That's because PopoverAnchorTooltip will only hide the tooltip if the children of the tooltip are the popover anchor ref element, which in our case is not because we wrap the tooltip to the inner animated view.
https://github.com/Expensify/App/blob/e9c112a2b89217401dfef3e248fa8e93d4c4e418/src/components/Tooltip/PopoverAnchorTooltip.tsx#L11-L16

What if we move the tooltip to the outer pressable, it will look like this:

https://github.com/Expensify/App/assets/50919443/0da9bd94-134b-4d7f-9b46-8fd794a2f1ba

@paultsimura @mountiny

File: src/pages/home/sidebar/ProfileAvatarWithIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                    tooltipText={translate('profilePage.profile')}
+                    fallbackIcon={currentUserPersonalDetails.fallbackIcon}
+                    isLoading={!!isLoading && !currentUserPersonalDetails.avatar}
+                />
+            </View>
+        </OfflineWithFeedback>
+    );
+}
+
+ProfileAvatarWithIndicator.displayName = 'PressableAvatarWithIndicator';
```
[the remark this answers is not in the document]
Comment (bernhardoj) [reply src/pages/home/sidebar/ProfileAvatarWithIndicator.tsx]: Updated

**Comment by shawnborton** [comment]
> What if we move the tooltip to the outer pressable, it will look like this:

Personally I would be okay with that, I think it's fine. Curious if @Expensify/design agrees!

**Comment by dubielzyk-expensify** [comment]
Yeah, I'm okay with that as well 👍

**Comment by bernhardoj** [comment]
@paultsimura Great. Updated the code and test.

**Comment by bernhardoj** [comment]
Oh, we got a TS error because of renaming the PressableAvatarWithIndicator to ProfileAvatarWithIndicator
image

What should we do here?

**Comment by mountiny** [comment]
no problem, continue with the review

**Review by paultsimura — APPROVED** [review]
LGTM.

I've also retested the second issue and attached the recording – ✔️

**Review by mountiny — APPROVED** [review]
thank you!

**Comment by mountiny** [comment]
The JS file was renamed

**Pull request merged**